netket/nn/symmetric_linear.py

```diff
@@ -242,41 +227,48 @@ def __call__(self, x: Array) -> Array:
 
         dtype = jnp.promote_types(x.dtype, self.dtype)
         x = jnp.asarray(x, dtype)
+        # Infer in_features and add batch and in_feature dimensions as necessary
+        if x.ndim == 1:
+            in_features = 1
+            x = jnp.expand_dims(x, (0, 1))
+        elif x.ndim == 2:
+            in_features = 1
+            x = jnp.expand_dims(x, 1)
+        else:
+            in_features = x.shape[-2]
 
-        x = (
-            x.reshape(-1, self.n_cells, self.sites_per_cell)
-            .transpose(0, 2, 1)
-            .reshape(-1, self.sites_per_cell, *self.shape)
-        )
+        x = x.reshape(*x.shape[:-1], self.n_cells, self.sites_per_cell)
+        x = x.transpose(0, 1, 3, 2)
+        x = x.reshape(*x.shape[:-1], *self.shape)
 
         # generate the default kernel init if necessary
         kernel_init = (
-            get_default_densesymm_init(self.n_cells * self.sites_per_cell)
+            get_default_densesymm_init(self.n_cells * self.sites_per_cell * in_features)
             if self.kernel_init is default_densesymm_initializer
             else self.kernel_init
         )
         kernel = self.param(
             "kernel",
             kernel_init,
-            (self.features, self.n_cells * self.sites_per_cell),
+            (self.features, in_features, self.n_cells * self.sites_per_cell),
             self.dtype,
         )
 
         kernel = jnp.asarray(kernel, dtype)
 
         if self.mask is not None:
-            kernel = kernel * jnp.expand_dims(self.mask, 0)
+            kernel = kernel * jnp.expand_dims(self.mask, (0, 1))
 
         kernel = self.make_kernel(kernel)
 
-        x = jnp.fft.fftn(x, s=self.shape).reshape(*x.shape[:2], self.n_cells)
+        x = jnp.fft.fftn(x, s=self.shape).reshape(*x.shape[:3], self.n_cells)
 
         kernel = jnp.fft.fftn(kernel, s=self.shape).reshape(
-            *kernel.shape[:3], self.n_cells
+            *kernel.shape[:4], self.n_cells
         )
 
         x = lax.dot_general(
-            x, kernel, (((1,), (2,)), ((2,), (3,))), precision=self.precision
+            x, kernel, (((1, 2), (1, 3)), ((3,), (4,))), precision=self.precision
         )
```

Comment:
I want to benchmark this stuff...
I fear this is suboptimal now..

Reply:
why exactly? 
we contract middle indices in `DenseEquivariant` too, there's no real way around it, FFTing on middle indices would be worse and harder to write
however, in `DenseEquivariant` the index list is `(((1, 2), (1, 2)), ((3,), (4,)))`, I must check why it's different

Reply:
> Should I transpose it to make it dimension 2? It's easy and needs to be done once in `setup`

This is now done